src/addons/Confirm/Confirm.js

```diff
@@ -1,68 +1,103 @@
 import _ from 'lodash'
-import React, { PropTypes } from 'react'
+import React, { Component, PropTypes } from 'react'
 
-import { getUnhandledProps, META } from '../../lib'
+import {
+  createShorthand,
+  customPropTypes,
+  getUnhandledProps,
+  META,
+} from '../../lib'
 import Button from '../../elements/Button'
 import Modal from '../../modules/Modal'
 
 /**
- * A Confirm modal gives the user a choice to confirm or cancel an action
+ * A Confirm modal gives the user a choice to confirm or cancel an action/
  * @see Modal
  */
-function Confirm(props) {
-  const { open, cancelButton, confirmButton, header, content, onConfirm, onCancel } = props
-  const rest = getUnhandledProps(Confirm, props)
-
-  // `open` is auto controlled by the Modal
-  // It cannot be present (even undefined) with `defaultOpen`
-  // only apply it if the user provided an open prop
-  const openProp = {}
-  if (_.has(props, 'open')) openProp.open = open
-
-  return (
-    <Modal {...openProp} size='small' onClose={onCancel} {...rest}>
-      {header && <Modal.Header>{header}</Modal.Header>}
-      {content && <Modal.Content>{content}</Modal.Content>}
-      <Modal.Actions>
-        <Button onClick={onCancel}>{cancelButton}</Button>
-        <Button primary onClick={onConfirm}>{confirmButton}</Button>
-      </Modal.Actions>
-    </Modal>
-  )
-}
+class Confirm extends Component {
+  static propTypes = {
+    /** The cancel button text. */
+    cancelButton: PropTypes.string,
 
-Confirm._meta = {
-  name: 'Confirm',
-  type: META.TYPES.ADDON,
-}
+    /** The OK button text. */
+    confirmButton: PropTypes.string,
 
-Confirm.propTypes = {
-  /** Whether or not the modal is visible */
-  open: PropTypes.bool,
+    /** The ModalContent text. */
+    content: customPropTypes.itemShorthand,
 
-  /** The cancel button text */
-  cancelButton: PropTypes.string,
+    /** The ModalHeader text. */
+    header: customPropTypes.itemShorthand,
 
-  /** The OK button text */
-  confirmButton: PropTypes.string,
+    /**
+     * Called when the Cancel button is clicked.
```

Comment:
Let's correct this, "Called when the Modal is closed without clicking confirm."  Since, it is called on Escape and click outside as well.